edb/lib/std/25-numoperators.eql

```diff
@@ -934,19 +1038,29 @@ std::`%` (n: std::decimal, d: std::decimal) -> std::decimal
 };
 
 
+# need an explicit operator for int64 in order to guarantee the result
+# is float64 and not decimal
 CREATE INFIX OPERATOR
-std::`^` (n: std::float64, p: std::float64) -> std::float64
-    FROM SQL OPERATOR '^';
+std::`^` (n: std::int64, p: std::int64) -> std::float64
+{
+    FROM SQL OPERATOR r'*';
+    FROM SQL 'SELECT (n ^ p)';
+};
 
 
 CREATE INFIX OPERATOR
-std::`^` (n: std::int64, p: std::int64) -> std::decimal
+std::`^` (n: std::float32, p: std::float32) -> std::float32
 {
     FROM SQL OPERATOR '*';
```

Comment:
Need a comment here clarifying why `^` is created from `*`.

Reply:
Right, and I should probably copy `/` instead of the `*` anyways because power behaves more like division than multiplication w.r.t. types and other meta-information. This is simply because `2^-2` is `1/4`.